pkg/ui/workspaces/e2e-tests/Dockerfile

```diff
@@ -0,0 +1,51 @@
+# WARNING:
+# This Dockerfile must be built from the root of cockroachdb.git with a prebuilt
+# cockroach binary at ./upstream_artifactsl/cockroach. Running it from other
+# directories or without a prebuilt cockroach binary will cause errors.
+#
+# Build (from the root of cockroach.git):
+# $ DOCKER_BUILDKIT=1 docker build \
+#     -f ./pkg/ui/workspaces/e2e-tests/Dockerfile \
+#     -t cockroachdb/cockroach-ci-ui \
+#     $PWD
+#
+# Run Locally (from the root of cockroach.git):
+# $ docker run \
+#     --rm \
+#     -it \
+#     -v $PWD:/upstream_artifacts \
+#     -v $PWD/pkg/ui/workspaces/e2e-tests/cypress:/cypress
+#     cockroachdb/cockroach-ci-ui
+#
+# Run in TeamCity (from the root of cockroach.git):
+# $ docker run \
+#     --rm \
+#     -it \
+#     -v $PWD/upstream_artifacts:/upstream_artifacts \
+#     -v $PWD/artifacts:/artifacts \
+#     cockroachdb/cockroach-ci-ui
+
+FROM cypress/browsers:node16.14.2-slim-chrome100-ff99-edge
```

Comment:
I'm pretty done with tests that build a Docker image every time they run. Building the image takes a long time and it's just a matter of time until something skews so things start failing. Can we pre-build this once and push it to our DockerHub?

Reply:
> Building the image takes a long time and it's just a matter of time until something skews so things start failing.

Well the good news is these are pretty static / minimal images.  The build here took 1m 28s in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_WipBaragCypressHealth/5936309?buildTab=log&focusLine=3&linesState=232, but that's still longer than the tests themselves.  We could probably build and publish an image that's `cypress/browsers:node16.14.2-slim-chrome100-ff99-edge` + `apt install libresolv-wrapper` only, then inject a `cockroach` binary and the `pkg/ui/workspaces/e2e-tests` tree at runtime and  run `yarn install && yarn test` inside the container.

When we want to test against newer browser versions in CI, we'd have to rebuild and republish that image but that shouldn't be so bad.  I'll message you about the mechanics of getting that image published!

Reply:
I mean, we could also go full-pancake and not maintain our own image at all:

```
  docker run \
    --rm \
    -it \
    -e 'IS_DOCKER=1' \
    -e 'COCKROACH=/upstream_artifacts/cockroach' \
    -v $PWD:/upstream_artifacts \
    -v $PWD/pkg/ui/workspaces/e2e-tests:/cypress \
    cypress/browsers:node16.14.2-slim-chrome100-ff99-edge \
    /bin/bash -c 'apt install libresolv-wrapper; pushd /cypress; yarn install --force; yarn test'
```

runs locally, which really isn't so bad 🤔

Reply:
I agree with @rickystewart. We shouldn't be relying on fetching docker images and ideally all other types of artifacts from the public internet. There is always a chance of that artifact being temporarily unavailable either due to a rate limit (e.g., [1]) or a transient (public) network issue; this will inevitably lead to flaky tests. Not to mention the security aspect... always a chance of a BGP spoofing attack to serve you rogue containers.


[1] https://docs.docker.com/docker-hub/download-rate-limit/#what-is-the-download-rate-limit-on-docker-hub